pkgs/development/python-modules/paddle-bfloat/default.nix

```
@@ -0,0 +1,43 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+, pythonAtLeast
+, numpy
+}:
+let
+  pname = "paddle-bfloat";
+  version = "0.1.7";
+in
+buildPythonPackage {
+  inherit pname version;
+  format = "setuptools";
+
+  src = fetchPypi {
+    pname = "paddle_bfloat";
+    inherit version;
+    hash = "sha256-mrjQCtLsXOvqeHHMjuMx65FvMfZ2+wTh1ao9ZJE+9xw=";
+  };
+
+  postPatch = ''
+    sed '1i#include <memory>' -i bfloat16.cc # gcc12
```

```suggestion
    sed '1i#include <memory>' -i bfloat16.cc # gcc12
    # replace deprecated function for python3.11
    substituteInPlace bfloat16.cc \
      --replace "Py_TYPE(&NPyBfloat16_Descr) = &PyArrayDescr_Type" "Py_SET_TYPE(&NPyBfloat16_Descr, &PyArrayDescr_Type)"
```
I don't know how upstream makes binaries for python 3.11, but without this patch, the build fails.
This function is implemented in python 3.9 and later.

https://docs.python.org/3.11/whatsnew/3.11.html#whatsnew311-c-api-porting

---

besides this, LGTM.